src/v/archival/ntp_archiver_service.cc

```diff
@@ -1194,7 +1194,8 @@ ss::future<ntp_archiver_upload_result> ntp_archiver::upload_segment(
           _conf->cloud_storage_initial_backoff,
           &rtc.get());
 
-        co_await _remote.upload_object(
+        // TODO(abhijat): Explain why it is safe to ignore index upload result.
```

Comment:
If we fail to upload the index but successfully upload the segment, the read path will create the index on the fly while downloading the segment, so it is okay to ignore the index upload failure, we still want to advance the offsets because the segment did get uploaded.